hpx/hpx_init_params.hpp

```diff
@@ -59,13 +59,15 @@ namespace hpx
         HPX_EXPORT void on_exit() noexcept;
         HPX_EXPORT void on_abort(int signal) noexcept;
         // Default params to initialize the init_params struct
-        extern HPX_EXPORT const hpx::program_options::options_description
-            default_desc;
         extern HPX_EXPORT startup_function_type default_startup;
         extern HPX_EXPORT shutdown_function_type default_shutdown;
         HPX_MAYBE_UNUSED extern HPX_EXPORT int dummy_argc;
         // TODO: make it only one parameter, probably add a cast
         HPX_MAYBE_UNUSED extern HPX_EXPORT char **dummy_argv;
+        static char app_name[] = HPX_APPLICATION_STRING;
```

Comment:
Let's maybe add a comment here to say why this one isn't in the cpp file.

Reply:
I think this could even be `constexpr` nowadays.

Reply:
Actually I get this when I try to make it a constexpr:
```
/src/hpx_init_params.cpp:18:31:error: cannot initialize an array element of type 'char *' with an lvalue of type 'const char [24]'
    char *default_argv[2] = { app_name, nullptr };
                              ^~~~~~~~
```